docs/docs/aztec/concepts/advanced/storage/note_discovery.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Note Discovery
+tags: [storage, concepts, advanced, notes]
+sidebar_position: 3
+---
+
+Note discovery refers to the process of a user identifying and decrypting the [encrypted notes](../../storage/notes.md) that belong to them.
+
+## Existing solutions
+
+### Brute force / trial-decrypt
+
+In some existing protocols, the user downloads all possible notes and tries to decrypt each one. If the decryption succeeds, the user knows they own that note. However, this approach becomes exponentially more expensive as the network grows and more notes are created. It also introduces a third-party server to gather and trial-decrypt notes, which is an additional point of failure. Note that this note discovery technique is not currently implemented for Aztec.
+
+### Off-chain communication
+
+Another proposed solution is having the sender give the note content to the recipient via some off-chain communication. While it solves the brute force issue, it introduces reliance on side channels which we don't want in a self-sufficient network. This option incurs lower transaction costs because fewer logs needs to be posted on-chain.
+
+## Aztec's solution: Note tagging
+
+Aztec introduces an approach that allows users to identify which notes are relevant to them by *tagging* the log in which the note is created. This is known as note tagging. The tag is generated in such a way that only the sender and recipient can identify it.
+
+### How it works
+
+#### Every log has a tag
+
+In Aztec, each emitted log is an array of fields, eg `[x, y, z]`. The first field (`x`) is a *tag* field used to index and identify logs. The Aztec node exposes an API `getLogsByTags()` that can retrieve logs matching specific tags.
+
+#### Tag generation
+
+The sender and recipient share a predictable scheme for generating tags. The tag is derived from a shared secret and an index (a shared counter that increments each time the sender creates a note for the recipient).
+
+#### Discovering notes in Aztec contracts
+
+This note discovery scheme is implemented by Aztec contracts rather than by the PXE. This means that users can update or use other types of note discovery to suit their needs.
```

Comment:
This is not true atm, though we will eventually achieve this, perhaps around June 2025.